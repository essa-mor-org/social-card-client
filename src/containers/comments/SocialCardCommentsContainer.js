import React, { PureComponent } from 'react';
import { connect } from 'react-redux';

import SocialCardComments from '../../components/social-card/comments/SocialCardComments';
import { makeGetPostCommentsSelector } from '../../selectors/Comments';
import { getPost } from '../../selectors/Posts';
import { commentOnPost } from '../../actions/Posts';

export class SocialCardCommentsContainer extends PureComponent {
    constructor(props) {
        super(props);
        this.onComment = this.onComment.bind(this);
    }

    onComment(comment) {
        const { id, commentOnPost, profileName, profileImage } = this.props;
        commentOnPost({ postId: id, profileName, profileImage, comment });
    }

    render() {
        const { postComments, profileImage } = this.props;
        return <SocialCardComments
            onComment={this.onComment}
            profileImage={profileImage}
            comments={postComments} />;
    }
}

const makeMapStateToProps = () => {
    const getPostCommentsSelector = makeGetPostCommentsSelector();
    const mapStateToProps = (state, ownProps) => {
        const post = getPost(state, ownProps).toJS();
        return {
            postComments: getPostCommentsSelector(state, ownProps),
            profileName: post.profileName,
            profileImage: post.profileImage
        }
    };
    return mapStateToProps
};



export default connect(makeMapStateToProps, { commentOnPost })(SocialCardCommentsContainer);