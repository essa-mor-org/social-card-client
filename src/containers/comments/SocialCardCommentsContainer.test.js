import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { SocialCardCommentsContainer } from './SocialCardCommentsContainer';

describe('SocialCardCommentsContainer test', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(<SocialCardCommentsContainer id={1} profileName="profileName" profileImage="profileImage" postComments={[]}/>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders correctly - comment callback', () => {
		const postId = 1,
			profileName = "profileText",
			profileImage = "profileImage",
			comment = "hi";

		const commentCallBack = jest.fn();
		const wrapper = mount(<SocialCardCommentsContainer
			id={postId}
			commentOnPost={commentCallBack}
			profileName={profileName}
			profileImage={profileImage}
			postComments={[]} />);
		const input = wrapper.find('.Social_comments_input');
		input.simulate('keyPress', { key: 'Enter', target: { value: 'hi' } });
		expect(commentCallBack).toHaveBeenCalledWith({ postId, profileName, profileImage, comment });
	});
});
