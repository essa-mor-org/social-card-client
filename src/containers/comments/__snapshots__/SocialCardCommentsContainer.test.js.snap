// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialCardCommentsContainer test renders correctly 1`] = `
<div
  className="Social_comments_input_panel"
>
  <img
    alt=""
    className="Social_comments_image"
    src="profileImage"
  />
  <input
    className="Social_comments_input"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Write a comment..."
    type="text"
    value=""
  />
</div>
`;
